components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -99,6 +99,74 @@ fn json_unquote(arg: &Option<Json>) -> Result<Option<Bytes>> {
     })
 }
 
+fn json_with_bytes_validator(expr: &tipb::Expr) -> Result<()> {
+    let children = expr.get_children();
+    assert!(children.len() >= 2);
+    // args should be like `&Option<Json> , &[&Option<Bytes>]`.
+    super::function::validate_expr_return_type(&children[0], EvalType::Json)?;
+    for i in 1..children.len() {
+        super::function::validate_expr_return_type(&children[i], EvalType::Bytes)?;
+    }
+    Ok(())
+}
+
+/// args should be like `&Option<Json> , &[&Option<Bytes>]`.
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_with_bytes_validator)]
+#[inline]
+fn json_extract(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    assert!(args.len() >= 2);
+    let j: &Option<Json> = args[0].as_ref();
+    let j = match j.as_ref() {
+        None => return Ok(None),
+        Some(j) => j.to_owned(),
+    };
+
+    let path_expr_list = match path_list(&args[1..])? {
+        Some(p) => p,
+        None => return Ok(None),
+    };
+
+    Ok(j.extract(&path_expr_list))
+}
+
+/// args should be like `&Option<Json> , &[&Option<Bytes>]`.
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_with_bytes_validator)]
+#[inline]
+fn json_remove(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    assert!(args.len() >= 2);
+    let j: &Option<Json> = args[0].as_ref();
+    let mut j = match j.as_ref() {
+        None => return Ok(None),
+        Some(j) => j.to_owned(),
+    };
+
+    let path_expr_list = match path_list(&args[1..])? {
+        Some(p) => p,
+        None => return Ok(None),
+    };
+
+    j.remove(&path_expr_list)?;
+    Ok(Some(j))
+}
+
+fn path_list(args: &[ScalarValueRef]) -> Result<Option<Vec<PathExpression>>> {
+    let mut path_expr_list = Vec::with_capacity(args.len());
+    for arg in args {
+        let json_path: &Option<Bytes> = arg.as_ref();
+
+        let json_path = match json_path.as_ref() {
+            None => return Ok(None),
+            Some(p) => String::from_utf8(p.to_vec()),
+        }
+        .map_err(crate::codec::Error::from)?;
+
+        let path_expr = parse_json_path_expr(&json_path)?;
+
+        path_expr_list.push(path_expr);
+    }
+    Ok(Some(path_expr_list))
```

Comment:
Looks like it will never be None?

Reply:
> Looks like it will never be None?

In line 158, if any `json_path` is None, it will return None.